lib/active_admin/orm/active_record/comments/comment.rb

```diff
@@ -4,7 +4,7 @@ class Comment < ActiveRecord::Base
     self.table_name = "#{table_name_prefix}active_admin_comments#{table_name_suffix}"
 
     belongs_to :resource, polymorphic: true
-    belongs_to :author,   polymorphic: true
+    belongs_to :author,   polymorphic: true, required: false
```

Comment:
Would it make sense to add a presence validation to the model conditional on whether authentication is enabled or not? That way we could keep the same integrity validation for the 90% case (authentication enabled).

Reply:
I don't quite remember the details, it looks like Rails has changed the default value for this option, so I have to set it to false.

Reply:
“ensures that no comments are created without an author” that's not true since I saw that there was a comment created by an "anonymous" author but not raising an exception.

Reply:
Yeah, but that would be the case when authentication is disabled, right? I'm proposing to add a conditional presence validation only when authentication is enabled.